### DISCUSSION TITLE
How to present a timer

### DISCUSSION THREAD

**Discussion by markusgermar** [ctreffe/alfred#228] [Alfred Q&A] [answered]
Hi there,

I would like to show participants a live timer (in sec and min) that show the time elapsed since the participant opened the current page.

Any idea?

Thank you!
Markus

**Comment by ctreffe** [comment]
Hey @markusgermar,

I would recommend using JavaScript to add a simple timer to your page. There are lots of examples on Stack Overflow for this. To give you an idea of how to implement this in alfred, I used a very basic example from [this question](https://stackoverflow.com/questions/37187504/javascript-second-counter):

```python
import alfred3 as al
exp = al.Experiment()

myPage = al.Page(title="Hello, world!", name="hello_world")

myPage += al.Text("You've been on this page for <span id=timer_seconds></span> seconds.")

myPage += al.JavaScript(
    '''
    window.setInterval((function(){
        var start = Date.now();
        var textNode = document.createTextNode('0');
        document.getElementById('timer_seconds').appendChild(textNode);
        return function() {
            textNode.data = Math.floor((Date.now()-start)/1000);
        };
    }()), 1000);
    '''
) 

exp += myPage

if __name__ == "__main__":
    exp.run()
```

The most important change I made in order to use this timer code with alfred is the split of HTML and JavaScript into two different alfred elements. In the above example, the HTML was added to the page as a text element, while the JavaScript code was added with our JavaScript element, which was designed for exactly this type of scenario.

There are more complex examples that you could tailor to your needs in [this second question on Stack Overflow](https://stackoverflow.com/questions/5517597/plain-count-up-timer-in-javascript). Among other topics, there is an example in which you can transfer the current timer state to another page. Most importantly, these examples contain code which separates the displayed time interval into seconds and minutes, as you requested.

**Comment by jobrachem** [comment, marked as answer]
You can also just use the CountUp element:
https://alfredo3.psych.bio.uni-goettingen.de/docs/generated/alfred3.element.display.CountUp.html

**Reply by markusgermar** [reply]
Thank you! That seems a little bit simpler than the solution of ctreffe. Is it possible to intergate the CountUp element in a text and/or change whether the time is presented in bold/italic letters?

**Reply by jobrachem** [reply]
I don't think you can easily place the timer inline. But you should be able to change the appearance with CSS in a `Style` element: https://alfredo3.psych.bio.uni-goettingen.de/docs/generated/alfred3.element.misc.Style.html?highlight=style#alfred3.element.misc.Style
